lib/cdo/aws/s3.rb

```diff
@@ -74,6 +84,12 @@ def self.create_client
     # @param [String] key
     # @return [String]
     def self.download_from_bucket(bucket, key, options = {})
+      if rack_env?(:development)
+        # For development environments, we look to see if we should lazily populate the
+        # local bucket first.
+        Cdo::LocalDevelopment.populate_local_s3_bucket(bucket, key)
+      end
```

Comment:
In the version of "idea 2" where aws_s3_emulated became a property in config.yml.erb rather than a function in cdo.rb, I think it'd be sane to key this feature directly off aws_s3_emulated. One of the "features" of CDO.aws_s3_emulated (if set) becomes populating the bucket, irregardless of environment.

Potentially this if/end clause is redundant anyway since `def populate_local_s3_bucket()` starts with the line:
```
return unless CDO.aws_s3_emulated?
```

Reply:
Also potentially relevant: would we want this behavior on rack_env?(:test) too when not in drone?

Reply:
We don't want this behavior in the test environment in the short term, but I could see us wanting it eventually. I think orienting around a `aws_s3_emulated` config option rather than anything environment-aware makes sense!

And yeah, the guard clause in `download_from_bucket` is indeed redundant with the much more important one in `populate_local_s3_bucket`, but I like the clarity of intent we get from including it explicitly :shrug: